src/data/perf_stat.rs

```diff
@@ -433,11 +434,12 @@ struct EndPerfData {
     pub metadata: GraphMetadata,
 }
 
-fn get_values(values: Vec<PerfStat>, key: String) -> Result<String> {
+fn get_values(values: Vec<PerfStat>, key: String, metrics: &mut DataMetrics) -> Result<String> {
```

Comment:
Since every data source is going to implement something like this, shouldn't we make a rust trait to hold some of the assumed common functionality every data source needs?